osf_oauth2_adapter/views.py

```diff
@@ -0,0 +1,76 @@
+import requests
+
+from django.views.generic.base import TemplateView
+
+from allauth.socialaccount.adapter import DefaultSocialAccountAdapter
+
+from allauth.utils import valid_email_or_none
+
+from allauth.account.utils import user_email, user_username, user_field
+
+from .apps import OsfOauth2AdapterConfig
+from allauth.socialaccount.providers.oauth2.views import OAuth2Adapter, OAuth2LoginView, OAuth2CallbackView
+
+from osf_oauth2_adapter.provider import OSFProvider
+
+
+class OSFOAuth2Adapter(OAuth2Adapter, DefaultSocialAccountAdapter):
+    provider_id = OSFProvider.id
+    base_url = '{}oauth2/{}'.format(OsfOauth2AdapterConfig.osf_accounts_url, '{}')
+    access_token_url = base_url.format('token')
+    authorize_url = base_url.format('authorize')
+    profile_url = '{}v2/users/me/'.format(OsfOauth2AdapterConfig.osf_api_url)
+
+    def populate_user(self, request, sociallogin, data):
+        """
+        Hook that can be used to further populate the user instance.
+
+        For convenience, we populate several common fields.
+
+        Note that the user instance being populated represents a
+        suggested User instance that represents the social user that is
+        in the process of being logged in.
+
+        The User instance need not be completely valid and conflict
+        free. For example, verifying whether or not the username
+        already exists, is not a responsibility.
+        """
+        username = data.get('username')
+        first_name = data.get('first_name')
+        last_name = data.get('last_name')
+        email = username
+        name = data.get('name')
+        time_zone = data.get('time_zone')
+        locale = data.get('locale')
+        gravatar = data.get('profile_image_url')
```

Comment:
Fixed this. Pulled the gravatar url, base64ed it and threw it in _identicon. I also added fields for time_zone and locale on the user and made sure those got where they were going.